### PR TITLE
Feature: use semver to parse version constraints

### LINKED ISSUE #20: Handle deep NPM dependencies
I think there should be a way to handle deep NPM dependencies (just like fossa does), for instance with a new optional parameter `--depth=2`.

### PR DESCRIPTION
**Pull request by nockty**
`semver` is an NPM module that can be used to check if a version satisfies a constraint. I think it's a good idea to use it in the `NpmPackageParser` to get the most accurate version for a dependency. It can also parse complex contraints such as *'1.x || >=2.5.0 || 5.0.0 - 7.2.3'*.

Closes #20 by adding a depth parameter for NPM dependencies.

### CODE REVIEW

**Comment by dkarchmer** [comment]
I don't know that I like this change. It introduces an ugly dependency to NPM (and nodeJS).
Any way to make this optional such that it is not a requirement. i.e. if you have it, you can select the option and get better results, but it works even without it.

**Comment by nockty** [comment]
So I made the usage of `semver` optional by implementing a simple Python parser for NPM versions.

I still have to update the README and remove `package.json` and `package-lock.json`.

File: utils/parsers/npm_parser.py [hunk 1]
```diff
@@ -20,45 +23,119 @@ def get_package_info(cls, name, version=None):
...
+        if use_semver:
+            # js2py isn't perfect so we treat special cases to prevent an infinite loop...
+            regex = r"(?P<maj>[0-9]+)\.(?P<min>[0-9]+)\.(?P<pat>[0-9]+)-?(?P<tag>[.a-z0-9]*)"
+            # if only a version is given in the spec, then only this version is valid
+            match = re.match(regex, version_spec)
+            if match is not None and match.span() == (0, len(version_spec)):
+                return version == version_spec
+            if version_spec == '^' + version:
+                return True
+            return semver.satisfies(version, version_spec)
```
Comment (dkarchmer) [inline utils/parsers/npm_parser.py]: What happens if semver is not installed? Do you check for that somewhere?

Comment (nockty) [reply utils/parsers/npm_parser.py]: I fixed the initialization of `semver` at the beginning of the file (as we don't want to do it if `use_semver` is false). However, it seems that `semver` is automatically installed by `js2py` so there is no need to check this.

Nevertheless, if one tries to run the script with `use_semver=true` without Node installed, one will get this: `AssertionError: You must have node installed! run: brew install node`. I think the error is explicit enough so we don't have to check this?

**Pull request merged**